### ISSUE TITLE
Build for ARM (e.g. Raspberry PI )

### ISSUE THREAD

**Issue by dtrehas** [dontpanic92/wxGo#4]
Dear Sir,

Your work is amazing.

Do you have any idea how this could be compiled in ARM?

Thank you in advance.

**Comment by dontpanic92** [comment]
Hi dtrehas,

I'm not familiar with ARM but I think it will be hard to compile wxGo on a Raspberry PI considering its limited performance. Perhaps we can cross-compile it. However I'm not sure whether it will work, because I have no experience on cross-compiling a Go program.

**Comment by neonsoftware** [comment]
Hi dontpanic92, 

first of all GREAT WORK 👍 thanks for sharing

Regarding cross compilation, it is quite beautiful and builtin in golang, [this post shows it very well](http://dave.cheney.net/2015/08/22/cross-compilation-with-go-1-5).
Basically it is mostly about playing with the variables GOOS and GOARCH.

here is a little example
```bash
# Let's say that hello.go prints "hello world". I'm on 64bit OSX
$ go build hello.go                          # compiles binary hello for my system
$ file hello                                 # checking binary type
hello: Mach-O 64-bit executable x86_64       # OK, it's mine

# now let's cross compile to linux arm
$ export GOOS=linux                          # I set 'linux as target OS' 
$ export GOARCH=arm                          # I set 'arm as target architecture'
$ go build hello.go                          # this time will cross-compile
$ file hello                                 # checking binary type
hello: ELF 32-bit LSB executable, ARM, version 1 (SYSV), statically linked, not stripped
# arm !
```

Regarding this specific project, cross-compiling this library to ARM, I guess that some specific architecture-specific assets will be required for the SWIG, wxWidget toolchain to compile. (a file setup_linux_arm.go ?)

I might have time to take a look in the following weeks. In that case will post updates here. 

cheers

**Comment by dontpanic92** [comment]
@neonsoftware First of all we need a wxWidgets for arm, and after that the `setup_linux_arm.go` should contain the `CPPFLAGS` and `LDFLAGS` to make it work.

**Comment by neonsoftware** [comment]
Yep 👍  I'll grab my arm board and try to compile the static wxWidgets libraries for arm

**Comment by neonsoftware** [comment]
@dontpanic92 did you have to manually set the static libraries to be PIC enabled with -fPIC ? 
I am having that sort of error ...

**Comment by neonsoftware** [comment]
been out for a week, will get back to it in the next days :bowtie:
